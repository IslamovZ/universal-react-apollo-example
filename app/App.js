/* eslint react/prefer-stateless-function: 0 */

import React from 'react';
import WOW from 'wow.js/dist/wow.js';

import 'flag-icon-css/css/flag-icon.css';
import { Header, Page } from 'components';
import './styles.css';

export default class App extends React.Component {

  componentDidMount() {
    new WOW().init();
  }

  render() {
    return (
      <div className="application">
        <Header />
        <Page />
      </div>
    );
  }
}
