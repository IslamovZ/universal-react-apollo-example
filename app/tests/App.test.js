import React from 'react';
import sinon from 'sinon';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import WOW from 'wow.js/dist/wow.js';

import App from 'App';

describe('Test suite for App Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <MockedProvider>
        <App />
      </MockedProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should init WOW.js in `componentDidMount`', () => {
    const component = (
      <MockedProvider>
        <App />
      </MockedProvider>
    );

    sinon.spy(App.prototype, 'componentDidMount');
    sinon.spy(WOW.prototype, 'init');
    mount(component);
    expect(App.prototype.componentDidMount.calledOnce).toBe(true);
    expect(WOW.prototype.init.calledOnce).toBe(true);
  });
});
