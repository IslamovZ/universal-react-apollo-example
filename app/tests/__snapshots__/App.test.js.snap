exports[`Test suite for App Component renders correctly 1`] = `
<div
  className="application">
  <header
    className={undefined}>
    <div
      className={undefined}
      data-wow-delay="0.5s">
      <h2>
        WeLikeGraphQL
      </h2>
    </div>
    <nav
      className={undefined}>
      <div>
        <div>
          Loading....
        </div>
        <div
          className="dropdown btn-group btn-group-default">
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={false}
            id="dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button">
            <span />
          </button>
          <ul
            aria-labelledby="dropdown"
            className="dropdown-menu"
            role="menu" />
        </div>
      </div>
      <div>
        Loading....
      </div>
    </nav>
  </header>
  <div>
    Loading....
  </div>
</div>
`;
